nemo/collections/nlp/models/dialogue/dialogue_gpt_classification_model.py

```diff
@@ -276,7 +344,7 @@ def forward(self, input_ids, attention_mask, labels, inference=True):
                     attn_mask,
                     labels=left_shifted_input_ids,
                     taskname_ids=prompt_ids,
-                    inference=inference,
+                    inference=False if len(prompt_ids.size()) == 2 else True,
```

Comment:
I think its better to have an inference flag that gets set in a way that clearly shows why inference is true or false. In this check, it is not immediately clear why `len(prompt_ids.size()) == 2 ` being true or false should indicate that we are doing inference.

Reply:
Sure, I agree it would be clearer that way